### PR TITLE
feat: Remove files marked as temporary

### PR DESCRIPTION
**Pull request by ivadym**
### This PR:

https://github.com/Clinical-Genomics/BALSAMIC/issues/1219

### Added:
- Snakemake flag to delete generated temporary files 

### Review and tests: 
- [ ] Tests pass
- [ ] Code review
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

File: tests/models/test_snakemake_models.py [hunk 1]
```diff
@@ -244,7 +244,7 @@ def test_get_snakemake_command(
     # THEN the expected format should be returned
     assert (
         snakemake_command
-        == f"snakemake --notemp -p --rerun-trigger mtime --directory {session_tmp_path.as_posix()} "
+        == f"snakemake --notemp -p --rerun-trigger --delete-temp-output mtime --directory {session_tmp_path.as_posix()} "
```
Comment (mathiasbio) [inline tests/models/test_snakemake_models.py:247]:
```suggestion
        == f"snakemake --notemp -p --rerun-trigger mtime --delete-temp-output  --directory {session_tmp_path.as_posix()} "
```

Comment (mathiasbio) [reply tests/models/test_snakemake_models.py:247]: I think mtime is the argument to the --rerun-trigger but maybe I missed something :D

File: BALSAMIC/models/snakemake.py [hunk 2]
```diff
@@ -166,7 +166,7 @@ def get_snakemake_options_command(self) -> str:
     def get_command(self) -> str:
         """Return Snakemake command to be submitted."""
         snakemake_command: str = (
-            f"snakemake --notemp -p --rerun-trigger mtime "
+            f"snakemake --notemp -p --rerun-trigger --delete-temp-output mtime "
```
Comment (mathiasbio) [inline BALSAMIC/models/snakemake.py:169]:
```suggestion
            f"snakemake --notemp -p --rerun-trigger mtime --delete-temp-output"
```

Comment (mathiasbio) [reply BALSAMIC/models/snakemake.py:169]: Same here!

**Review by mathiasbio — APPROVED** [review]
Nicely done! Thanks for going through these temp file : ) 
Just these comments...but feel free to just resolve if they're nonsense!

File: tests/models/test_snakemake_models.py (hunk 1 above)
Comment (ivadym) [reply tests/models/test_snakemake_models.py:247]: Oh good catch! Totally a moment of blindness, thanks!

**Comment by ivadym** [comment]
Was nice to dream about this, but it's nos possible to change how temp files are handled if we use `--immediate-submit`

`--immediate-submit has to be combined with --notemp, because temp file handling is not supported in this mode.` 


Closing this PR and the associated issue 🙃

**Pull request closed without merging**